pennylane/gradients/vjp.py

```diff
@@ -151,7 +151,7 @@ def compute_vjp_single(dy, jac, num=None):
 
         # Single measurement with dimension e.g. probs
         else:
-            jac = qml.math.stack(jac)
+            jac = qml.math.reshape(qml.math.stack(jac), (-1, num))

```

Comment:
This is an excellent catch! It seems that the JVP calculation also suffers from this (not detected by the bug, but similar code logic). Would you be interested in fixing this as well?
- Minimal failing example is the test I suggested below but replace `batch_vjp` by `batch_jvp`.
- I'd expect/hope that a fix is a similar one-line change

Reply:
Hi, I am not sure about this, so I don't know if I can fix this. If anyone else wants to give it a shot, please go ahead :)

Reply:
Nevermind, I actually had made a different mistake when looking at the `batch_jvp` example, and the bug is not present for JVPs :)